pkg/controller/bastion/actuator.go

```diff
@@ -53,12 +53,12 @@ func (a *actuator) getAWSClient(ctx context.Context, bastion *extensionsv1alpha1
 		return nil, fmt.Errorf("failed to find %q Secret: %w", v1beta1constants.SecretNameCloudProvider, err)
 	}
 
-	credentials, err := aws.ReadCredentialsSecret(secret, false)
+	authConfig, err := aws.ReadCredentialsSecret(secret, false)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read credentials Secret: %w", err)
 	}
-
-	return awsclient.NewClient(string(credentials.AccessKeyID), string(credentials.SecretAccessKey), shoot.Spec.Region)
+	authConfig.Region = shoot.Spec.Region
```

Comment:
This is only a nit, but since we do this in a couple places:

Can't the secret get the function above `ReadCredentialsSecret` get an optional `Region` param ? We basically have to do this everywhere, except for DNSRecords where it's the only place actually to have a region in the secret. For all other cases the objects (like a shoot, has a region in it's own spec).

Reply:
Yes, makes sense. I will make the change